### ISSUE TITLE
assert self.dll_directory is not None, module_name AssertionError: scipy

### ISSUE THREAD

**Issue by hanshantong** [Nuitka/Nuitka#2112]
Before submitting an Issue, please review the [Issue Guidelines](https://github.com/Nuitka/Nuitka/blob/develop/CONTRIBUTING.md#submitting-an-issue).

* Please check whether the bug was already reported or fixed.
Was reported ([here](https://github.com/Nuitka/Nuitka/issues/1934)) but still broken.

* Please check out if the develop version of Nuitka works better for you.
1.5 or 1.6rc2 both exhibit the same issue.
  
* Nuitka version, full Python version, flavor, OS, etc. as output by *this*
  command.

nuitka info:
1.5
Commercial: None
Python: 3.8.0 (default, Nov  6 2019, 16:00:02) [MSC v.1916 64 bit (AMD64)]
Flavor: Anaconda Python
Executable: C:\Anaconda3\envs\latestNuitka\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10

* How did you install Nuitka and Python
Using anaconda3 and creating a python virtual env, using pip to install the latest nuitka from pypi.

* The specific PyPI names and versions

accessible-pygments==0.0.3
alabaster==0.7.13
Babel==2.12.1
beautifulsoup4==4.11.2
certifi @ file:///C:/b/abs_85o_6fm0se/croot/certifi_1671487778835/work/certifi
charset-normalizer==3.1.0
colorama==0.4.6
contourpy==1.0.7
cycler==0.11.0
delvewheel==1.3.1
docutils==0.19
fonttools==4.39.0
idna==3.4
imageio==2.26.0
imagesize==1.4.1
importlib-metadata==6.0.0
importlib-resources==5.12.0
Jinja2==3.1.2
kiwisolver==1.4.4
MarkupSafe==2.1.2
matplotlib==3.6.0
Nuitka==1.5
numpy==1.24.2
numpydoc==1.5.0
opencv-python==4.7.0.72
ordered-set==4.1.0
packaging==23.0
pandas==1.5.3
pefile==2023.2.7
Pillow==9.4.0
psutil==5.9.4
pydata-sphinx-theme==0.13.1
Pygments==2.14.0
pyparsing==3.0.9
PySide6==6.4.2
PySide6-Addons==6.4.2
PySide6-Essentials==6.4.2
python-dateutil==2.8.2
pytz==2022.7.1
PyYAML==6.0
requests==2.28.2
scipy==1.10.1
seaborn==0.12.2
sentry-sdk==1.16.0
shapely==2.0.1
shiboken6==6.4.2
six==1.16.0
snowballstemmer==2.2.0
soupsieve==2.4
Sphinx==6.1.3
sphinx-book-theme==1.0.0
sphinx-remove-toctrees==0.0.3
sphinxcontrib-applehelp==1.0.4
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==2.0.1
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.5
thop==0.1.1.post2209072238
torch==1.13.1
torchvision==0.14.1
tqdm==4.65.0
typing_extensions==4.5.0
ultralytics==8.0.53
urllib3==1.26.15
wincertstore==0.2
zipp==3.15.0
zstandard==0.20.0

* Many times when you get an error from Nuitka, your setup may be special
My code works normally when run in PyCharm with right mouse key option "Run 'program_name'".

* Also supply a [Short, Self Contained, Correct, Example](https://sscce.org/)

```python
from ultralytics import YOLO
model = YOLO(model='yolov8n.pt', task='segment')
images = r"D:\Projects\algorithms\x_ray_void_hole\void_20230227\tmp"
results = model(source=images, project='./results', save=True, save_txt=False)
print(results)
```

Trying to compile it with:
```python
python -m nuitka --standalone --enable-plugin=matplotlib  YoLoPackageSnippet.py
```
failed with:
```python
PASS 1: 59.7%|██████████████▉          | 596/998, pandas.core.nanopsTraceback (most recent call last):
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\__main__.py", line 151, in <module>
    main()
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\__main__.py", line 137, in main
    MainControl.main()
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\MainControl.py", line 1088, in main
    _main()
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\MainControl.py", line 950, in _main
    main_module = _createMainModule()
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\MainControl.py", line 244, in _createMainModule
    optimizeModules(main_module.getOutputFilename())
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\optimizations\Optimization.py", line 337, in optimizeModules
    finished = makeOptimizationPass()
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\optimizations\Optimization.py", line 293, in makeOptimizationPass
    changed = optimizeModule(current_module)
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\optimizations\Optimization.py", line 180, in optimizeModule
    changed = optimizeCompiledPythonModule(module)
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\optimizations\Optimization.py", line 140, in optimizeCompiledPythonModule
    considerUsedModules(module=module, signal_change=signalChange)
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\importing\Recursion.py", line 448, in considerUsedModules
    _addParentPackageUsages(
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\importing\Recursion.py", line 407, in _addParentPackageUsages
    used_package_module = recurseTo(
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\importing\Recursion.py", line 84, in recurseTo
    module, added_flag = _recurseTo(
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\importing\Recursion.py", line 43, in _recurseTo
    module, is_added = Building.buildModule(
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\tree\Building.py", line 1262, in buildModule
    ) = readSourceCodeFromFilenameWithInformation(
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\tree\SourceHandling.py", line 182, in readSourceCodeFromFilenameWithInformation
    source_code_modified, contributing_plugins = Plugins.onModuleSourceCode(
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\plugins\Plugins.py", line 911, in onModuleSourceCode
    new_source_code = plugin.onModuleSourceCode(module_name, source_code)
  File "C:\Anaconda3\envs\latestNuitka\lib\site-packages\nuitka\plugins\standard\DelvewheelPlugin.py", line 103, in onModuleSourceCode
    assert self.dll_directory is not None, module_name
AssertionError: scipy
```
* Provide in your issue the Nuitka options used
```python
python -m nuitka --standalone --enable-plugin=matplotlib  YoLoPackageSnippet.py
```
**However, an exception is that the nuitka version 1.1 can compile normally, and generate executable file.** But when run the executable file, error emerges as listed below:
```python
Traceback (most recent call last):
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\YoLoPackageSnippet.py", line 14, in <module>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\ultralytics\__init__.py", line 5, in <module ultralytics>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\ultralytics\yolo\__init__.py", line 3, in <module ultralytics.yolo>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\ultralytics\yolo\v8\__init__.py", line 3, in <module ultralytics.yolo.v8>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\ultralytics\yolo\v8\classify\__init__.py", line 3, in <module ultralytics.yolo.v8.classify>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\ultralytics\yolo\v8\classify\predict.py", line 3, in <module ultralytics.yolo.v8.classify.predict>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\torch\__init__.py", line 990, in <module torch>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\torch\_meta_registrations.py", line 16, in <module torch._meta_registrations>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\torch\_refs\__init__.py", line 14, in <module torch._refs>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\torch\_prims\__init__.py", line 31, in <module torch._prims>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\torch\_subclasses\__init__.py", line 3, in <module torch._subclasses>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\torch\_subclasses\fake_tensor.py", line 14, in <module torch._subclasses.fake_tensor>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\torch\fx\__init__.py", line 84, in <module torch.fx>  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\torch\fx\_symbolic_trace.py", line 1077, in <module torch.fx._symbolic_trace>
  File "D:\Projects\PyQtProjects\QtProjects\YoLoPackageSnippet.dist\torch\fx\_symbolic_trace.py", line 1009, in wrap
NotImplementedError: wrap must be called at the top level of a module
```

**Comment by kayhayen** [comment]
Torch will be better with newer Nuitka. Since you are using Anaconda, can you tell me, what is from conda environment, and what is from PyPi there?

**Comment by hanshantong** [comment]
> Torch will be better with newer Nuitka. Since you are using Anaconda, can you tell me, what is from conda environment, and what is from PyPi there?
I create virtual pyton environment using command 'conda create -n env_name python==version_xxx', this will download some packages from anaconda website, as listed below:
```python
# Name                    Version                   Build  Channel
ca-certificates           2023.01.10           haa95532_0    defaults
certifi                   2022.12.7        py38haa95532_0    defaults
openssl                   1.1.1t               h2bbff1b_0    defaults
pip                       23.0.1           py38haa95532_0    defaults
python                    3.8.0                hff0d562_2    defaults
setuptools                65.6.3           py38haa95532_0    defaults
sqlite                    3.41.1               h2bbff1b_0    defaults
vc                        14.2                 h21ff451_1    defaults
vs2015_runtime            14.27.29016          h5e58377_2    defaults
wheel                     0.38.4           py38haa95532_0    defaults
wincertstore              0.2              py38haa95532_2    defaults

```
and other packages in the virtual are installed by pip. for example, "PySide6, ultralytics, matplotlib, numpy, pandas, shapely, opencv, nuitka, delvewheel, imageio" and so on.
**Note**: the 'torch' package  comes with the pacakge 'ultralytics'.

**Comment by kayhayen** [comment]
I see, most Anaconda users seem to use the conda form of scipy for performance. I have a similar issue for "numpy" giving an error with anaconda, that I have yet to see. Let me try scipy in that way.

**Comment by kayhayen** [comment]
I have so far failed to reproduce this, compiling scipy in a conda env with pip install and a pure virtualenv under anaconda python. 

May I ask you to debug that code in `DelvewheelPlugin` yourself? It should not be too hard. I don't reproduce it for some reason.

**Comment by hanshantong** [comment]
> 

Can you run and package the snippet code I provided before? Is there any debug tutorial using DelvewheelPlugin? I just found some usages on its website. I'll learn and try it.

**Comment by hanshantong** [comment]
> 



> I have so far failed to reproduce this, compiling scipy in a conda env with pip install and a pure virtualenv under anaconda python.
> 
> May I ask you to debug that code in `DelvewheelPlugin` yourself? It should not be too hard. I don't reproduce it for some reason.

Can you run and package the snippet code I provided before? Is there any debug tutorial using DelvewheelPlugin? I just found some usages on its website. I'll learn and try it.

**Comment by hanshantong** [comment]
> 

I wrap the snippet code into a wheel file and using 'delvewheel show my_wheel_name.whl' the wheel file, No any dll file is listed. Then I using 'pip install my_wheel_name.whl' into my anaconda env and it can run normally. I don't know yet where is the problem when packaging the code using nuitka? It is very so weird.

**Comment by kayhayen** [comment]
Somehow I thought this was about scipy, but I am not sure now. Your last comment doesn't make any sense to me. I don't think that is what I asked. I added a reporting to the plugin mechanism that should make it more clear generally. Also can you reduce your test case to "import scipy" and see the issue.

**Comment by hanshantong** [comment]
> Somehow I thought this was about scipy, but I am not sure now. Your last comment doesn't make any sense to me. I don't think that is what I asked. I added a reporting to the plugin mechanism that should make it more clear generally. Also can you reduce your test case to "import scipy" and see the issue.

OK, I just try to package the following code, it can package and run normally:
```python
import scipy
def main():
    print(scipy.__version__)
if __name__ == "__main__":
    main()
```

**Comment by kayhayen** [comment]
What the plugin does is to take part of the source code of scipy and execute it in a known environment, with `__file__` set, so it can be observed to do `os.add_dll_directory`. It's kind of surprising for this to work in one compilation, but not the other. Unfortunately you decided to strip the output, so I cannot even tell what version of delvewheel is detected in scipy.

So again, I invite you to check the code of delvewheel for yourself, and figure this out. I will try your reproducer only once I find the time myself.

**Comment by hanshantong** [comment]
> What the plugin does is to take part of the source code of scipy and execute it in a known environment, with `__file__` set, so it can be observed to do `os.add_dll_directory`. It's kind of surprising for this to work in one compilation, but not the other. Unfortunately you decided to strip the output, so I cannot even tell what version of delvewheel is detected in scipy.
> 
> So again, I invite you to check the code of delvewheel for yourself, and figure this out. I will try your reproducer only once I find the time myself.

The compilation output is so short:
```python
Nuitka-Options:INFO: Used command line options: --standalone main.py
Nuitka:INFO: Starting Python compilation with Nuitka '1.5' on Python '3.8' commercial grade 'not installed'.
Nuitka:INFO: Completed Python level compilation and optimization.               
Nuitka:INFO: Generating source code for C backend compiler.
Nuitka:INFO: Running data composer tool for optimal constant value handling.                 
Nuitka:INFO: Running C compilation via Scons.
Nuitka-Scons:INFO: Backend C compiler: cl (cl 14.2).
Nuitka-Scons:INFO: Backend linking program with 106 files (no progress information available).
Nuitka-Scons:INFO: Compiled 106 C files using clcache with 105 cache hits and 1 cache misses.
Nuitka-Plugins:WARNING: dll-files: 
DLL configuration by filename code for 'numpy' did not give a result. Either conditions are missing, or this version of the module needs treatment added.
Nuitka-Plugins:INFO: dll-files: Found 1 file DLLs from numpy installation.
Nuitka-Plugins:INFO: dll-files: Found 33 files DLLs from scipy installation.
Nuitka:INFO: Keeping build directory 'main.build'.                                                                                          
Nuitka:INFO: Successfully created 'main.dist\main.exe'.
```
and I run the executable file 'main.exe' in command line, it outputs the scipy's version:
```python
1.9.1
```

**Comment by hanshantong** [comment]
> I mean, the less warnings that you ignored, I get to see, the better, right?

Got it,  I just try the latest nuitka version (1.5.3), it can compile and run the following snippet code normally:
```python
from ultralytics import YOLO
def main():
    model = YOLO(model='yolov8n.pt', task='segment')
    images = r"my_project_path\tmp"
    results = model(source=images, project='./results', save=True, save_txt=False)
    print(results)

if __name__ == "__main__":
    main()
```

but in my another code that also contains 'ultralytics' library, two warrnings are printed:
```python
Nuitka-Plugins:WARNING: pyside6: Unwanted import of 'tkinter' that is redundant with 'PySide6' encountered. Use '--nofollow-import-to=tkinter' or uninstall it for best compatibility with pure Python execution.

Nuitka-Plugins:WARNING: dll-files: DLL configuration by filename code for 'numpy' did not give a result. Either conditions are missing, or this version of the module needs treatment added.

```
And when analyzing depenencies, there is a new error:
```python
Nuitka-Inclusion:INFO: Analyzing dependencies of 'my_project_path\MainWindow.exe'.
Nuitka:INFO: Running 'depends.exe' for my_project_path\MainWindow.dist\MainWindow.exe took 0.80 seconds

Traceback (most recent call last):
  File "C:\Anaconda3\envs\package6.4.2\lib\site-packages\nuitka\__main__.py", line 151, in <module>
    main()
  File "C:\Anaconda3\envs\package6.4.2\lib\site-packages\nuitka\__main__.py", line 137, in main
    MainControl.main()
  File "C:\Anaconda3\envs\package6.4.2\lib\site-packages\nuitka\MainControl.py", line 1090, in main
    _main()
  File "C:\Anaconda3\envs\package6.4.2\lib\site-packages\nuitka\MainControl.py", line 993, in _main
    detectUsedDLLs(
  File "C:\Anaconda3\envs\package6.4.2\lib\site-packages\nuitka\freezer\Standalone.py", line 734, in detectUsedDLLs
    _detectUsedDLLs(
  File "C:\Anaconda3\envs\package6.4.2\lib\site-packages\nuitka\freezer\Standalone.py", line 721, in _detectUsedDLLs
    addIncludedEntryPoint(dll_entry_point)
  File "C:\Anaconda3\envs\package6.4.2\lib\site-packages\nuitka\freezer\IncludedEntryPoints.py", line 179, in addIncludedEntryPoint
    if areSamePaths(entry_point.dest_path, standalone_entry_point.dest_path):
  File "C:\Anaconda3\envs\package6.4.2\lib\site-packages\nuitka\utils\FileOperations.py", line 108, in areSamePaths
    path2 = getExternalUsePath(path2)
  File "C:\Anaconda3\envs\package6.4.2\lib\site-packages\nuitka\utils\FileOperations.py", line 986, in getExternalUsePath
    filename = getWindowsShortPathName(filename)
  File "C:\Anaconda3\envs\package6.4.2\lib\site-packages\nuitka\utils\FileOperations.py", line 951, in getWindowsShortPathName
    raise WindowsError(
FileNotFoundError: [Errno 2] system not found specific file。

```

**Comment by kayhayen** [comment]
I see, that sheds a good light on things. The numpy is from handling the crash in another issue, but that change might have poisoned `numpy` from conda packages, but you are not using that. For you, this should have happened all the time, or you should have had the crash this is fixing. 

The new crash is based on a wrong assumption in hotfix code. I guess, this needs more work to get right.

**Comment by hanshantong** [comment]
> For you, this should have happened all the time, or you should have had the crash this is fixing.

Thanks, kayhayen. How should I address that.

**Comment by hanshantong** [comment]
> Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html
> 
> Please retry with this, I believe it should be fixed. If that fails, please provide the file created by `--report=compilation-report.xml` which should allow me get more information.

Hi, kayhayen. I have tried the development version you advised. It works. The compilation progress just shot two WARNINGS:
```python
Nuitka-Plugins:WARNING: pyside6: Unwanted import of 'tkinter' that is redundant with 'PySide6' encountered. Use '--nofollow-import-to=tkinter' or uninstall it for best compatibility with pure Python execution.

Nuitka-Plugins:WARNING: dll-files: DLL configuration by filename code for 'numpy' did not give a result. Either conditions are missing, or this version of the module needs treatment added.
```
and my program can be run normally.

Besides, I want to report some small issues. My program is wrote by pyside6. It uses QtSQL series classes to access MySQL 8.0.11. When finishing compilation, the nuitka  does not copy the "sqldriver" in the corresponding plugin diretory. In my case, the "sqldrivers" is at "C:\Anaconda3\envs\my_env_name\Lib\site-packages\PySide6\plugins\sqldrivers". 

And a dll file named "libmysql.dll" which MUST be needed if one wants to connect to MySQL is not copied to the ".../program_name/dist/PySide6/" directory.

And a package called "ultralytics" that is located at "C:\Anaconda3\envs\my_env_name\Lib\site-packages\ultralytics" is not copied to ".../program_name/dist" directory.

**Comment by kayhayen** [comment]
@hanshantong please make these new issues. I am closing this early, because it never became too clear why we were working on anyway, I do not think seeing it will help anybody in the future. The numpy warning is expected for now with what I will cann pseudo-Anaconda for now. Having a configuration detected that it is not only anaconda Python, but in fact the Anaconda package of numpy, will be resolved separately, once I figure out how to detect that for my reporting, and testing anyway.

**Comment by kayhayen** [comment]
This is part of the hotfix release 1.5.4 that I just made.

**Comment by hanshantong** [comment]
> This is part of the hotfix release 1.5.4 that I just made.

Got it, thanks kayhayen.

**Comment by hanshantong** [comment]
> > This is part of the hotfix release 1.5.4 that I just made.
> 
> Got it, thanks kayhayen.

@kayhayen I just try the version 1.5.4, when it is closed to finish compiling, an error about "upix" was ocurring:
```python
FATAL: upx: Error, call to 'path/upx-4.0.2-win64/upx.exe' failed: ['path/upx-4.0.2-win64/upx.exe', '-q', '--no-progress', 'my_project_path\\test\\main.dist\\libcrypto-1_1-x64.dll'] -> b'upx: my_project_path\\test\\main.dist\\libcrypto-1_1-x64.dll: AlreadyPackedException: already packed by UPX\r\n'.


```
